### PR TITLE
Support python 3.12 and vtk 9.3.0 in CI

### LINKED ISSUE #5195: docstring for `regular_faces` now fails with vtk 9.3.0
### Describe what maintenance you would like added.

vtk 9.3.0 just was published on pypi.   This docstring now fails:

```txt
856         Get the face array of a tetrahedron as a 4x3 array
857 
858         >>> import pyvista as pv
859         >>> tetra = pv.Tetrahedron()
860         >>> tetra.regular_faces
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,4 @@
    -array([[0, 1, 2],
    -       [1, 3, 2],
    -       [0, 2, 3],
    -       [0, 3, 1]])
    +array([[0, 2, 1],
    +       [1, 2, 3],
    +       [0, 3, 2],
    +       [0, 1, 3]])
```

It is due to a change in the ordering of points/faces in vtk.  See https://github.com/Kitware/VTK/commit/3aab7579cb78c5bf8c2208c46727c106338078af

### Links to source code.

_No response_

### Pseudocode or Screenshots

_No response_

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

Support testing python 3.12 and latest vtk in CI (9.3.0).

### Details

Closes #5195

### CODE REVIEW

File: .github/workflows/testing-and-deployment.yml [hunk 1]
```diff
@@ -186,7 +188,7 @@ jobs:
         run: pip install 'numpy<1.24'
 
       - name: Limit Matplotlib for VTK<=9.2.2
-        if: ${{ matrix.vtk-version != 'latest' && matrix.vtk-version != '9.2.2' }}
+        if: ${{ matrix.vtk-version == '9.1' || matrix.vtk-version != '9.0.3' }}
```
Comment (MatthewFlamm) [inline .github/workflows/testing-and-deployment.yml:191]:
```suggestion
        if: ${{ matrix.vtk-version == '9.1' || matrix.vtk-version == '9.0.3' }}
```

File: pyvista/core/pointset.py [hunk 2]
```diff
@@ -853,15 +853,16 @@ def regular_faces(self) -> np.ndarray:  # numpydoc ignore=RT01
 
         Examples
         --------
-        Get the face array of a tetrahedron as a 4x3 array
+        Get the regular face array of a plane with 2x2 arrangement of cells
+        as a 4x4 array.
 
         >>> import pyvista as pv
-        >>> tetra = pv.Tetrahedron()
```
Comment (MatthewFlamm) [inline pyvista/core/pointset.py:859]: This needed to be removed, see #5195.

File: .github/workflows/testing-and-deployment.yml [hunk 3]
```diff
@@ -142,6 +142,8 @@ jobs:
           - python-version: '3.10'
             vtk-version: '9.2.2'
           - python-version: '3.11'  # Added with VTK 9.2.5
```
Comment (tkoyama010) [inline .github/workflows/testing-and-deployment.yml:144]:
```suggestion
          - python-version: '3.11'
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thanks for this!

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**